Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1

```diff
@@ -0,0 +1,113 @@
+Class ThresholdRule
+{
+	[string]$counterInstanceName
+	[bool]$isHigherBetter
+	[float]$thresholdValue
+}
+
+function ValidateThresholdRule($cltAccountUrl,$headers,[ThresholdRule]$thresholdRule,$testRunId)
+{
+ $counterInstances = Get-CounterInstances $cltAccountUrl $headers $testRunId
+ return ValidateThreshold $cltAccountUrl $headers $thresholdRule $testRunId $counterInstances
+}
+
+function ValidateThreshold($cltAccountUrl,$headers,[ThresholdRule]$thresholdRule,$testRunId,$counterInstances)
+{
+ $counterInstanceId = GetCounterInstanceId $counterInstances $thresholdRule.counterInstanceName
+ $samplesRemaining = $true
+ $startInterval = 0
+ $endInterval = 2000
+ $violationCount = 0
+ While($samplesRemaining -eq $true)
+ {
+  $samples = Get-Samples $cltAccountUrl $headers $testRunId $counterInstanceId $startInterval $endInterval
+  if($samples -and $samples.count -gt 0)
+  {
+   $newViolations =  GetViolationsCount $samples $thresholdRule.thresholdValue $thresholdRule.isHigherBetter
+   $violationCount = $violationCount + $newViolations
+  }
+  if($samples.count -ne 2000)
```

Comment:
>($samples.count -ne 2000) [](start = 4, length = 25)

What is the strategy here to come out of the loop? Where are you accumulating the count of samples after each iteration of the while loop? What is this magic number 2000 to break the loop?